Attribute Analysis.ipynb

```diff
@@ -0,0 +1,650 @@
+    "        transforms.ToTensor()\n",
+    "        ])\n",
+    "dataset = datasets.CoCoDataset(transform_train)\n",
+    "folder_name = 'coco_example'\n",
+    "\n",
+    "# dataset = datasets.OpenImagesDataset(transform_train)\n",
+    "# folder_name = 'openimages_supp'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "save_loc = '1_pager_gen'\n",
+    "os.system(\"rm -r results/{0}/{1}\".format(folder_name, save_loc))\n",
+    "file = folder(save_loc, folder_name)\n",
+    "first_pass = True\n",
+    "to_write = {}\n",
+    "if not os.path.exists(\"checkpoints/{}\".format(folder_name)):\n",
+    "    os.mkdir(\"checkpoints/{}\".format(folder_name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_folder = None\n",
+    "\n",
+    "distances = pickle.load(open(\"results/{}/3.pkl\".format(folder_name), \"rb\"))\n",
+    "sample_file = distances[0][0][0][3]\n",
+    "if not os.path.exists(sample_file):\n",
+    "    assert data_folder is not None, \"initialize data_folder with folder path of your data\"\n",
+    "    dataset.init_folder_path(data_folder)\n",
+    "    print(\"overwriting from_path() function\")\n",
+    "    dataset.from_path = dataset.from_path_prerun"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Attribute\n",
+    "Fill in below with information about attribute you would like to analyze\n",
+    "<a id=\"attribute\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "attr_names = dataset.attribute_names\n",
+    "num_attrs = len(attr_names)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# (M1) Metric: Distance from center, size, attribute label inference\n",
+    "<a id=\"metric1\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "Fill in below with number of attribute values (e.g. 2 possible values), the assigned labels names, and which attribute you want the metric to focus on (e.g. female)\n",
+    "<a id=\"metric1_setup\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hide_toggle(for_next=True, toggle_text='Show/hide M1 code')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "info = pickle.load(open(\"results/{}/1.pkl\".format(folder_name), \"rb\"))\n",
+    "sizes = info['sizes']\n",
+    "dists = info['distances']\n",
+    "tiny_sizes = info['tiny_sizes']\n",
+    "no_faces = info['noface_sizes']\n",
+    "        \n",
+    "scenes = [None]*num_attrs\n",
+    "for attr in range(num_attrs):\n",
+    "    try:\n",
```

Comment:
does this try-except block every come up? also, we had discussed not printing messages like "tiny sizes is empty for [attribute]", and in the screenshots it never did, so maybe this try-except isn't necessary? please debug and if it can be removed without breaking things, remove the try-except